flocker/control/httpapi.py

```diff
@@ -853,6 +865,74 @@ def replace_configuration(self, applications, deployment):
         except ConfigurationError as e:
             raise make_bad_request(code=BAD_REQUEST, description=unicode(e))
 
+    @app.route("/configuration/leases", methods=['GET'])
+    # This can stop being private as part of FLOC-2741:
+    @private_api
+    @structured(
+        inputSchema={},
+        outputSchema={
+            '$ref':
+            '/v1/endpoints.json#/definitions/list_leases'
+        },
+        schema_store=SCHEMAS
+    )
+    def list_leases(self):
+        """
+        List the current leases in the configuration.
+        """
+        now = datetime.fromtimestamp(self.clock.seconds(), UTC)
+        result = []
+        for lease in self.persistence_service.get().leases.values():
+            result.append(lease_to_json(lease, now))
+        return result
+
+    @app.route("/configuration/leases/<dataset_id>", methods=['DELETE'])
+    # This can stop being private as part of FLOC-2741:
+    @private_api
+    @user_documentation(
+        u"""
+        This will release a lease on a dataset, allowing it to be moved
+        elsewhere, deleted and otherwise modified.
+        """,
+        header=u"Delete a lease from the configuration",
+        examples=[
+            u"delete a lease",
+        ],
+        section=u"dataset",
+    )
+    @structured(
+        inputSchema={},
+        outputSchema={'$ref': '/v1/endpoints.json#/definitions/lease'},
+        schema_store=SCHEMAS
+    )
+    def delete_lease(self, dataset_id):
+        """
+        Remove a lease from the cluster configuration.
+
+        :param unicode dataset_id: The dataset whose lease is being
+            removed.
+
+        :return: A ``Deferred`` firing with an ``EndpointResponse`` or
+            serializable JSON.
+        """
+        now = datetime.fromtimestamp(self.clock.seconds(), UTC)
+        leases = self.persistence_service.get().leases
+        dataset_id = UUID(dataset_id)
```

Comment:
This could throw a `ValueError` in the event of `dataset_id` in the request path not being a valid UUID.